src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -63,6 +76,17 @@ private enum Flags : byte
         private SynthesizedSealedPropertyAccessor _lazySynthesizedSealedAccessor;
         private CustomAttributesBag<CSharpAttributeData> _lazyCustomAttributesBag;
 
+#nullable enable
+        /// <summary>
+        /// A null value means we haven't yet calculated the backing field for field keyword.
+        /// A non-null value with a null <see cref="SynthesizedBackingFieldSymbol.ForFieldKeyword.BackingField" /> means we calculated the backing field, but there was none.
+        /// </summary>
+        /// <remarks>
+        /// This should only be set via CreateBackingFieldForFieldKeyword to avoid race conditions.
+        /// </remarks>
+        private SynthesizedBackingFieldSymbol.ForFieldKeyword? _lazyBackingFieldSymbolForFieldKeyword;
```

Comment:
>_lazyBackingFieldSymbolForFieldKeyword [](http://example.com/codeflow?start=63&length=38)

This feels like a move to the wrong direction. It looks like now we can end up with two backing field symbols at the same time. One is created for the initializer, another is created for the ```field``` keyword. Not only we increase the amount of information we store for a property symbol, we also create confusion with two distinct field symbols to work with and choose between. 
I assume that this change was made to be able to distinguish tree states:
1. We haven't created a backing field and we don't know if we need it. 
2. We haven't created a backing field, but we know that we don't need it. 
3. We created a backing field.

As I indicated in one of the earlier threads, we can distinguish all tree states with just one field by using a sentinel value that indicates the "unknown" state (null value become a "known" state when the field is not needed). Here is the quote: "Usually, we don't use any additional flags in order to be able to track three states like that. Instead, we use sentinel values to represent uninitialized/unknown state. See ```SourceMemberMethodSymbol._lazyIteratorElementType```, for example."

Given that it is hard to create a dummy value of ```SynthesizedBackingFieldSymbol```. I suggest doing the following:
- Add ```static private readonly _sentinel = new object();```
- Change ```_lazyBackingFieldSymbol``` as follows ```private object _lazyBackingFieldSymbol = _sentinel;```.
- The field value should be updated only if its value is equal to ```_sentinel``` value. ```InterlockedCompareExchange``` can help with that.
- A simple check against ```_sentinel``` value (instead of a null value) can be used to test if we are still in an unknown state and have some work to do.
- We might need to record an extra information about whether the symbol was created by constructor. That can be recorder in the ```SynthesizedBackingFieldSymbol``` the same way we record whether it was created for a ```field``` keyword.